### PR TITLE
Document use of AFL.

### LINKED ISSUE #3812: Document how to use solfuzzer with afl

### PR DESCRIPTION
**Pull request by chriseth**
Closes #3812

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -170,6 +170,54 @@ and re-run the test. It will now pass again:
...
+
+Next, you need some example source files. This will make it much easer for the fuzzer
+to find errors. You can either use the syntax test files or extract test files
+from the documentation or the other tests:
+
+::
+
+    mkdir /tmp/test_cases
+    cd /tmp/test_cases
+    path/to/solidity/scripts/extract_test_cases.py path/to/solidity/test/libsolidity/SolidityEndToEndTest.cpp
```
Comment (axic) [inline docs/contributing.rst]: Can you add the other example for extracting from documentation?

**Comment by chriseth** [comment]
Updated.

**Review by axic — DISMISSED** [review]

File: docs/contributing.rst (hunk 1 above)
Comment (chriseth) [reply docs/contributing.rst]: Done.

**Comment by axic** [comment]
Merging as `circlec/docs` succeeded.

File: docs/contributing.rst [hunk 2]
```diff
@@ -170,6 +170,57 @@ and re-run the test. It will now pass again:
...
+from the documentation or the other tests:
+
+::
+
+    mkdir /tmp/test_cases
+    cd /tmp/test_cases
+    # extract from tests:
+    path/to/solidity/scripts/isolate_tests.py path/to/solidity/test/libsolidity/SolidityEndToEndTest.cpp
+    # extract from documentation:
+    path/to/solidity/scripts/isolate_tests.py docs path/to/solidity/docs
```
Comment (axic) [inline docs/contributing.rst]: The arguments actually must be swapped.

**Review by axic — APPROVED** [review]

File: docs/contributing.rst [hunk 3]
```diff
@@ -170,6 +170,57 @@ and re-run the test. It will now pass again:
...
+    # extract from documentation:
+    path/to/solidity/scripts/isolate_tests.py path/to/solidity/docs docs
+
+The AFL documentation states that the corpus (the initial input files) should not be
+too large. The files themselves should not be larger than 1 kB and there should be
+at most one input file per functionality, so better start with a small number of
+input files. There is also a tool called ``afl-cmin`` that can trim input files
+that result in similar behaviour of the binary.
+
+Now run the fuzzer (the ``-m`` extends the size of memory to 200 MB):
```
Comment (axic) [inline docs/contributing.rst]: Is it 200 when the value below says 100?

Comment (axic) [inline docs/contributing.rst]: Is it 200 when the value below says 100?

**Review by axic — CHANGES_REQUESTED** [review]

File: docs/contributing.rst (hunk 3 above)
Comment (chriseth) [reply docs/contributing.rst]: yeah... I tried it with 300, let me check again what a currently good value is.

**Comment by chriseth** [comment]
Updated.

**Review by axic — DISMISSED** [review]

**Review by axic — APPROVED** [review]

**Pull request merged**